react-components/src/hooks/useAll3dModels.ts

```diff
@@ -0,0 +1,19 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+
+import { type CogniteClient, type HttpError, type Model3D } from '@cognite/sdk';
+
+const fetchModels = async (sdk: CogniteClient): Promise<Model3D[]> => {
+  return await sdk.models3D.list().autoPagingToArray({ limit: Infinity });
+};
+
+export function useAll3dModels(sdk: CogniteClient, enabled: boolean): UseQueryResult<Model3D[]> {
+  return useQuery<Model3D[], HttpError>({
+    queryKey: ['models'],
+    queryFn: async () => await fetchModels(sdk),
```

Comment:
Can't we call the endpoint directly here?

Reply:
https://github.com/cognitedata/reveal/pull/4746/commits/f51740e72efab7b49df11e4dc584ae056644e8f9